src/main/java/Containers/Portfolio.java

```diff
@@ -23,67 +23,116 @@ public static Portfolio getInstance() {
         return Portfolio.commonPortfolio;
     }
 
-    // Storage: The UUID of asset and the reference of asset
-    private final HashMap<String, Asset> storage;
+
+
+    // assetList: a list of asset in the portfolio
+    private List<Asset> assetList;
+    // transactionList: a list of transaction in the portfolio
+    private List<Transaction> transactionList;
```

Comment:
Consider changing naming to transactionHistory. What do others think about this?

Reply:
I agree, transactionhistory is a better name for this.